src/iiif/tools.py

```diff
@@ -364,3 +382,88 @@ def send_email(email_address, email_subject, email_body):
         html_content=email_body
     )
     sg.send(email)
+
+
+def check_zip_payload(payload):
+    if 'urls' not in payload or len(payload['urls']) == 0:
+        raise ImmediateHttpResponse(response=HttpResponse("No urls detected in json", status=400))
+
+
+def strip_full_iiif_url(url):
+    if '/iiif/' not in url:
+        raise ImmediateHttpResponse(response=HttpResponse("Misformed paths", status=400))
+
+    # Strip the domain from the url and return the only relevant part
+    return url.split('/iiif/')[-1]
+
+
+def store_zip_job(zip_info):
+    # There's a lot of things in the request_meta that is not JSON serializable.
+    # Since we just need some headers we simply remove all values that are not strings
+    zip_info['request_meta'] = {k: v for k, v in zip_info['request_meta'].items() if type(v) is str}
+
+    collection = Collection.objects.get(name=settings.ZIP_COLLECTION_NAME)
+    message = Message.objects.create(
+        raw_data=json.dumps(zip_info),
+        collection=collection
+    )
+    return message
+
+
+def create_tmp_folder():
+    zipjob_uuid = uuid4()
+    tmp_folder_path = f'/tmp/{zipjob_uuid}/'
+    os.mkdir(tmp_folder_path)
+    return zipjob_uuid, tmp_folder_path
+
+
+def save_file_to_folder(folder, filename, content):
+    with open(folder+filename, 'w') as f:
+        f.write(content)
+
+
+def create_local_zip_file(zipjob_uuid, folder_path):
+    zip_file_path = f'/tmp/{zipjob_uuid}.zip'
+    with ZipFile(zip_file_path, 'w') as zip_obj:
+        for filename in glob.glob(f"{folder_path}/*"):
+            zip_obj.write(filename)
+    return zip_file_path
+
+
+def get_object_store_connection():
+    return Connection(**settings.OBJECT_STORE)
+
+
+def store_object_on_object_store(connection, local_zip_file_path, filename):
+    with open(local_zip_file_path, 'r') as local:
+        connection.put_object(
+            settings.OBJECT_STORE_CONTAINER_NAME,
+            filename,
+            contents=local.read(),
+            content_type='application/zip'
+        )
```

Comment:
You should really use the available library, even though its minimal. Then you can add the creation of tmp urls there too, which is a nice feature.

Reply:
Requested access. Waiting for that..